src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/TasksEmailBuilder.java

```diff
@@ -66,78 +51,33 @@ private boolean checkProperties(Properties properties) {
         return SapsPropertiesUtil.checkProperties(properties, propertiesSet);
     }
 
-    private boolean checkFields(String userEmail, List<String> tasksId, PermanentStorage permanentStorage) {
-        return userEmail.trim().isEmpty() || Objects.isNull(userEmail) ||
-                tasksId.isEmpty() || Objects.isNull(tasksId) || Objects.isNull(permanentStorage);
+    private boolean checkFields(String userEmail, List<TaskEmail> tasksEmail) {
+        return userEmail.trim().isEmpty() || Objects.isNull(userEmail) || Objects.isNull(tasksEmail);
     }
 
     @Override
     public void run() {
         try {
-            JSONArray tasks = generateAllTasksJsonArray();
+            String tasksEmail = generateTasksEmail();
 
-            LOGGER.debug("Tasks JSON array: " + tasks.toString());
+            LOGGER.debug("Tasks JSON array: " + tasksEmail);
 
-            GoogleMail.Send(noReplyEmail, noReplyPass, userEmail, EMAIL_TITLE, tasks.toString(2));
-        } catch (JSONException e) {
-            LOGGER.error("Error while working with an object of the JSON type", e);
+            sendEmail(tasksEmail);
```

Comment:
These methods has only two lines of implementation, move code of them to here and deleted the methods.

Reply:
Done